src/pages/studyView/StudyViewUtils.tsx

```diff
@@ -2512,3 +2496,76 @@ export function updateSavedUserPreferenceChartIds(
     }
     return chartSettings;
 }
+
+export function getMolecularProfileOptions(
+    molecularProfiles: MolecularProfile[],
+    molecularPorfileSampleSet: { [id: string]: string[] }
+): GenomicDataCountWithSampleUniqueKeys[] {
+    return _.chain(molecularProfiles)
+        .groupBy(molecularProfile =>
+            molecularProfile.molecularProfileId.replace(
+                molecularProfile.studyId + '_',
+                ''
+            )
+        )
+        .map((profiles, value) => {
+            const uniqueProfiledsamples = _.chain(profiles)
+                .flatMap(
+                    molecularProfile =>
+                        molecularPorfileSampleSet[
+                            molecularProfile.molecularProfileId
+                        ] || []
+                )
+                .uniq()
+                .value();
+            return {
+                value: value,
+                label: profiles[0].name,
+                count: uniqueProfiledsamples.length,
+                sampleUniqueKeys: uniqueProfiledsamples,
+            };
+        })
+        .value();
+}
+
+export function getCNAsamplesCount(molecularPorfileSampleCountSet: {
+    [id: string]: GenomicDataCount;
+}) {
+    let count = 0;
+    if (molecularPorfileSampleCountSet['cna'] !== undefined) {
+        count = molecularPorfileSampleCountSet['cna'].count;
+    } else if (molecularPorfileSampleCountSet['gistic'] !== undefined) {
+        count = molecularPorfileSampleCountSet['gistic'].count;
+    } else if (molecularPorfileSampleCountSet['rae'] !== undefined) {
+        count = molecularPorfileSampleCountSet['rae'].count;
+    } else if (molecularPorfileSampleCountSet['cna_consensus'] !== undefined) {
+        count = molecularPorfileSampleCountSet['cna_consensus'].count;
+    }
```

Comment:
Should we do a for loop instead?

Reply:
may be not. I think order is important here